xbmc/peripherals/EventScanner.cpp

```diff
@@ -95,6 +95,10 @@ void CEventScanner::Process(void)
     const double nowMs = static_cast<double>(SystemClockMillis());
     const double scanIntervalMs = GetScanIntervalMs();
 
+    // Handle wrap-around
+    if (nowMs < nextScanMs)
+      nextScanMs = nowMs;
```

Comment:
The floating point quantity has an integer source, `SystemClockMillis()`, which uses unsigned int and can wrap around, so this check is needed.

I used doubles not because we're working with times, but because we're working with rates. This means that after reciprocating to get time, we're dealing with rational numbers and would only lose precision by casting earlier than the call to Sleep()